### PR TITLE
Chef 17 updates

### PR DESCRIPTION
**Pull request by ramereth**
- Chef 17 updates: enable `unified_mode` on all resources
- Bump required Chef Infra Client to >= 15.3
- Remove support and testing for RHEL 6 and Ubuntu 16.04

Signed-off-by: Lance Albertson <lance@osuosl.org>

### CODE REVIEW

File: spec/client_spec.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 describe 'rsyslog::client' do
   cached(:chef_run) do
-    ChefSpec::ServerRunner.new(platform: 'ubuntu', version: '16.04') do |node|
+    ChefSpec::ServerRunner.new(platform: 'ubuntu', version: '20.04') do |node|
```
Comment (tas50) [inline spec/client_spec.rb:5]: You can save yourself some pain and just remove the version here so you don't need to update it again

File: spec/default_spec.rb [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 describe 'rsyslog::default' do
   cached(:chef_run) do
-    ChefSpec::ServerRunner.new(platform: 'ubuntu', version: '16.04').converge(described_recipe)
+    ChefSpec::ServerRunner.new(platform: 'ubuntu', version: '20.04').converge(described_recipe)
```
Comment (tas50) [inline spec/default_spec.rb:5]: same here

File: spec/default_spec.rb [hunk 3]
```diff
@@ -13,7 +13,7 @@
 
   context "when node['rsyslog']['relp'] is true" do
     cached(:chef_run) do
-      ChefSpec::ServerRunner.new(platform: 'ubuntu', version: '16.04') do |node|
+      ChefSpec::ServerRunner.new(platform: 'ubuntu', version: '20.04') do |node|
```
Comment (tas50) [inline spec/default_spec.rb:16]: and here

File: spec/default_spec.rb [hunk 4]
```diff
@@ -26,7 +26,7 @@
   context "when node['rsyslog']['enable_tls'] is true" do
     context "when node['rsyslog']['tls_ca_file'] is not set" do
       cached(:chef_run) do
-        ChefSpec::ServerRunner.new(platform: 'ubuntu', version: '16.04') do |node|
+        ChefSpec::ServerRunner.new(platform: 'ubuntu', version: '20.04') do |node|
```
Comment (tas50) [inline spec/default_spec.rb:29]: and here

File: spec/default_spec.rb [hunk 5]
```diff
@@ -38,7 +38,7 @@
 
     context "when node['rsyslog']['tls_ca_file'] is set" do
       cached(:chef_run) do
-        ChefSpec::ServerRunner.new(platform: 'ubuntu', version: '16.04') do |node|
+        ChefSpec::ServerRunner.new(platform: 'ubuntu', version: '20.04') do |node|
```
Comment (tas50) [inline spec/default_spec.rb:41]: and here

File: spec/default_spec.rb [hunk 6]
```diff
@@ -50,7 +50,7 @@
 
       context "when protocol is not 'tcp'" do
         cached(:chef_run) do
-          ChefSpec::ServerRunner.new(platform: 'ubuntu', version: '16.04') do |node|
+          ChefSpec::ServerRunner.new(platform: 'ubuntu', version: '20.04') do |node|
```
Comment (tas50) [inline spec/default_spec.rb:53]: and here

**Review by bmhughes — APPROVED** [review]

**Comment by kitchen-porter** [comment]
Released as: [9.0.0](https://github.com/sous-chefs/rsyslog/releases/tag/9.0.0)

**Pull request merged**